src/utils/DateUtils.ts

```diff
@@ -18,18 +17,38 @@ export const toLocalDateTimeString = (date: number): string => {
 };
 
 export const toLocalDateString = (date: number): string => {
-  date = makeEpochMilli(date);
-  return new Date(date).toLocaleDateString(Localization.getLocale(), DATE_FORMAT_OPTIONS);
+  return new Date(formatDate(date)).toLocaleDateString(Localization.getLocale(), DATE_FORMAT_OPTIONS);
 };
 
-export const makeEpochMilli = (epoch: number): number => {
+const formatDate = (date: number): number => {
+  let epoch: number = date
+  epoch = formatFractionalPart(epoch)
   if (!isEpochMilli(epoch)) {
-    epoch = epoch * EPOCH_MILLISECONDS;
+    epoch = makeEpochMilli(epoch)
   }
-  return epoch;
+
+  return epoch
+}
+
+const makeEpochMilli = (date: number): number => {
+  if (!isEpochMilli(date)) {
+    date = date * EPOCH_MILLISECONDS;
+  }
+  return date;
+};
+
+const isEpochMilli = (date: number): boolean => {
+  return date.toString().length > 10;
 };
 
-export const isEpochMilli = (epoch: number): boolean => {
-  const epochLength = epoch.toString().length;
-  return epochLength > 10;
+const formatFractionalPart = (date: number): number => {
+  if (date.toString().includes('.')) {
+    const epochParts: Array<string> = date.toString().split('.')
+    const epoch: string = epochParts[0]
+    const fractionalPadded: string = epochParts[1].padEnd(3, '0');
+
+    return Number(`${epoch}${fractionalPadded}`)
```

Comment:
Hmzz, not sure whether this is always doing the correct thing. Can't we should just drop the fractions altogether here? We are interested in the data anyway, not the exact time.

We have code to determine whether it is milliseconds or seconds. So the fraction could be about milliseconds or seconds as well. I think the code assumes seconds for the fractional number and then creates milliseconds for the result because it simply adds them to the number

Reply:
Seems to have the same effect when the fractional part is removed. So I removed it.